### PR TITLE
#168 add option to choose first rendered page

### PR DESCRIPTION
**Pull request by micmro**
added `selectedPage` as new option.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -96,8 +96,17 @@
             //Source: http://www.webpagetest.org/result/151226_X7_b43d35e592fab70e0ba012fe11a41020/
             // window["fetch"]("test-data/www.bbc.co.uk.160529_8V_7R3.har")
             // window["fetch"]("test-data/www.bbc.com.har")
```
Comment (micmro) [inline src/index.html:98]: diversify our dev examples a bit.

File: src/ts/helpers/parse.ts [hunk 2]
```diff
@@ -143,9 +144,33 @@ export function escapeHtml(unsafe: string | number | boolean = ""): string {
...
+    return input;
+  } else if (typeof input === "string") {
     return parseInt(input, 10);
   } else {
-    return input;
+    return undefined;
   }
 }
+
+/** Validates the `ChartOptions` attributes types */
```
Comment (micmro) [inline src/ts/helpers/parse.ts:156]: do some basic type checking to make it easier for users to find small issues.

File: src/ts/paging/paging.ts [hunk 3]
```diff
@@ -80,7 +82,7 @@ export default class Paging {
     // remove all existing options, like placeholders
     removeChildren(selectbox);
     this.doc.pages.forEach((p, i) => {
-      let option = new Option(p.title, i.toString(), i === this.selectedPageIndex);
+      let option = new Option(p.title, i.toString(), false, i === this.selectedPageIndex);
```
Comment (micmro) [inline src/ts/paging/paging.ts:85]: I had this wrong: the third option was `defaultSelected` not `selected`.

> If an option is selected by default, it is displayed first in the drop-down list.

File: src/ts/paging/paging.ts [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 export default class Paging {
-  private selectedPageIndex = 0;
   private onPageUpdateCbs: OnPagingCb[] = [];
 
-  constructor(private doc: WaterfallDocs) {
-
+  constructor(private doc: WaterfallDocs, private selectedPageIndex = 0) {
+    if (selectedPageIndex >= this.doc.pages.length) {
+      // fall back to last item if doc has too few pages.
+      this.selectedPageIndex = this.doc.pages.length - 1;
```
Comment (micmro) [inline src/ts/paging/paging.ts:12]: Fall back to last item if doc has too few pages.

I think we should not throw an error for this - this is most likely what the user wanted anyway.
What do you think?

Comment (soulgalore) [reply src/ts/paging/paging.ts:12]: sounds good!

**Pull request merged**